src/main/java/com/github/hcsp/encapsulation/Cat.java

```diff
@@ -18,7 +21,13 @@
      * @param name 名字
      * @return 创建的猫
      */
-    public static Cat newCat(String name, int age) {}
+    public static Cat newCat(String name, int age) {
+        if (age<0 || name.equals("") || name == null) {
```

Comment:
如果`name`是空指针，这里会出问题。
```suggestion
        if (age<0 || name == null || name.equals("")) {
```

Reply:
没及时看到呢，不过后来也自己发现啦， 谢谢！